src/Pelagos/Entity/DatasetSubmission.php

```diff
@@ -1764,14 +1760,55 @@ public function getDatasetFileColdStorageArchiveSize()
         return $this->datasetFileColdStorageArchiveSize;
     }
 
+    /**
+     * Set the original filename of the archive file to be stored in cold storage.
+     *
+     * @param string|null $datasetFileColdStorageOriginalFilename The original filename to be preserved.
+     *
+     * @return void
+     */
+    protected function setDatasetFileColdStorageOriginalFilename($datasetFileColdStorageOriginalFilename)
+    {
+        $this->datasetFileColdStorageOriginalFilename = $datasetFileColdStorageOriginalFilename;
+    }
+
+    /**
+     * Get the original filename of the archive file stored in cold storage.
+     *
+     * @return string|null
+     */
+    public function getDatasetFileColdStorageOriginalFilename()
+    {
+        return $this->datasetFileColdStorageOriginalFilename;
+    }
+
+    /**
+     * Set the cold storage attributes, as one action, no nulls allowed.
+     *
+     * @param integer $filesize The original file size, in bytes, to be preserved.
+     * @param string  $hash     The original file sha256 hash to be preserved.
+     * @param string  $filename The original file name to be preserved.
+     *
+     * @return void
+     */
+    public function setDatasetFileColdStorageAttributes(int $filesize, string $hash, string $filename)
```

Comment:
I'm not sure whether this is a good practice of grouping setters, we haven't done it in the past (for ex: spatial extent)